remote_install_checkbox_snaps/remote_install_checkbox_snaps.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python3
+"""
+The checkbox snap also needs to install a runtime snap, which is specific to
+the OS version of the image is running. We try to select the correct runtime
+based on the track of checkbox snap being installed. If the runtime snap is
+specified then we install that one instead.
+"""
+import argparse
+import logging
+import shlex
+import subprocess
+import sys
+from dataclasses import dataclass
+from typing import List, Optional
+
+logging.basicConfig(level=logging.INFO)
+
+
+@dataclass
+class CheckboxOptions:
+    checkbox_snap: str
+    checkbox_channel: str
+    checkbox_track: str
+    checkbox_args: str
+    checkbox_runtime: Optional[str] = None
+
+
+class SSHClient:
+    def __init__(self, host: str, username: str):
+        self.host: str = host
+        self.username: str = username
+
+    def execute_command(self, command: str):
+        logging.info("Executing command on %s: %s", self.host, command)
+        try:
+            ssh_command = (
```

Comment:
True, and I agree these are probably options that we want to add here especially since we use them everywhere by default. However I don't think we can depend on the _run macro for this.

Reply:
I do see what you mean, but I think it might limit the utility of having this as a reusable script. Having it also handle the installation means that we can replace a good chunk of the scripting and templating for this in the jenkins job with just a command that we pass a few arguments to. In reality, we run everything with those same ssh args, so I don't see much need for them to change.

I'm curious if @jocave also has any thoughts on this, since I think it was his suggestion to put this in a reusable script?